src/Criteria.php

```diff
@@ -54,6 +56,20 @@ public static function expr()
         return self::$expressionBuilder;
     }
 
+    /**
+     * Merges two Criteria together.
+     *
+     * @return self
+     */
+    public static function merge(Criteria $leftCriteria, Criteria $rightCriteria)
+    {
+        return self::create()
+            ->andWhere(self::expr()->andX(...array_filter([$leftCriteria->getWhereExpression(), $rightCriteria->getWhereExpression()])))
+            ->orderBy(array_merge($leftCriteria->getOrderings(), $rightCriteria->getOrderings()))
+            ->setFirstResult($rightCriteria->getFirstResult() ?? $leftCriteria->getFirstResult())
+            ->setMaxResults($rightCriteria->getMaxResults() ?? $leftCriteria->getFirstResult());
```

Comment:
@TheDutchScorpion, is it a type or is it intended?
```suggestion
            ->setMaxResults($rightCriteria->getMaxResults() ?? $leftCriteria->getMaxResults());
```